plugins/node/opentelemetry-instrumentation-knex/src/utils.ts

```diff
@@ -45,7 +45,8 @@ export const getFormatter = (runner: any) => {
 
 export const cloneErrorWithNewMessage = (err: Exception, message: string) => {
   if (err && err instanceof Error) {
-    const clonedError = new (err.constructor as Exception)(message);
+    const clonedError = Object.assign({}, err);
+    clonedError.message = message;
```

Comment:
What do you think about the following change:

```diff
diff --git a/plugins/node/opentelemetry-instrumentation-knex/src/instrumentation.ts b/plugins/node/opentelemetry-instrumentation-knex/src/instrumentation.ts
index c80ebbe5..d539126f 100644
--- a/plugins/node/opentelemetry-instrumentation-knex/src/instrumentation.ts
+++ b/plugins/node/opentelemetry-instrumentation-knex/src/instrumentation.ts
@@ -176,8 +176,8 @@ export class KnexInstrumentation extends InstrumentationBase {
             const formatter = utils.getFormatter(this);
             const fullQuery = formatter(query.sql, query.bindings || []);
             const message = err.message.replace(fullQuery + ' - ', '');
-            const clonedError = utils.cloneErrorWithNewMessage(err, message);
-            span.recordException(clonedError);
+            const exc = utils.otelExceptionFromKnexError(err, message);
+            span.recordException(exc);
             span.setStatus({ code: api.SpanStatusCode.ERROR, message });
             span.end();
             throw err;
diff --git a/plugins/node/opentelemetry-instrumentation-knex/src/utils.ts b/plugins/node/opentelemetry-instrumentation-knex/src/utils.ts
index 12ba70ed..e3c82f78 100644
--- a/plugins/node/opentelemetry-instrumentation-knex/src/utils.ts
+++ b/plugins/node/opentelemetry-instrumentation-knex/src/utils.ts
@@ -14,17 +14,14 @@
  * limitations under the License.
  */
 
+import { Exception } from '@opentelemetry/api';
 import {
   DBSYSTEMVALUES_SQLITE,
   DBSYSTEMVALUES_POSTGRESQL,
 } from '@opentelemetry/semantic-conventions';
 
-type Exception = {
-  new (message: string): Exception;
-  constructor: Exception;
-  errno?: number;
+type KnexError = Error & {
   code?: string;
-  stack?: string;
 };
 
 export const getFormatter = (runner: any) => {
@@ -43,16 +40,17 @@ export const getFormatter = (runner: any) => {
   return () => '<noop formatter>';
 };
 
-export const cloneErrorWithNewMessage = (err: Exception, message: string) => {
-  if (err && err instanceof Error) {
-    const clonedError = Object.assign({}, err);
-    clonedError.message = message;
-    clonedError.code = err.code;
-    clonedError.stack = err.stack;
-    clonedError.errno = err.errno;
-    return clonedError;
+export function otelExceptionFromKnexError(err: KnexError, message: string): Exception {
+  if (!(err && err instanceof Error)) {
+    return err;
   }
-  return err;
+
+  return {
+    message,
+    code: err.code,
+    stack: err.stack,
+    name: err.name,
+  };
 };
 
 const systemMap = new Map([
```

This clarifies that the utility function is about returning an object to be used in `span.recordException`, i.e. type Exception (https://github.com/open-telemetry/opentelemetry-js/blob/main/api/src/common/Exception.ts). Benefits:

- It doesn't bother copying the `errno` field, which wasn't being used. (If Knex errors can include an errno instead of a 'code', we migth want to consider doing something with the errno field.) It also doesn't bother copying other err fields, if any.
- It also includes `err.name`, so there is a fallback value for the `exception.type` attribute if `err.code` is not defined. Otherwise `Span#recordException` can fail to record the exception: https://github.com/open-telemetry/opentelemetry-js/blob/01a2c35a694e57df45f063d61506ef9e9938eb7d/packages/opentelemetry-sdk-trace-base/src/Span.ts#L308-L329

This could all also be done in a separate change. I'm happy to PR that separately if you prefer.